src/lib/comms/sol-lwm2m-client.c

```diff
@@ -2487,10 +2541,48 @@ sol_lwm2m_client_start(struct sol_lwm2m_client *client)
 
     //Try to register with all available [non-bootstrap] servers
     SOL_VECTOR_FOREACH_IDX (&ctx->instances, instance, i) {
+        bool secure = true;
+        //Setup DTLS parameters
         r = read_resources(client, ctx, instance, res, 1,
-            SECURITY_SERVER_IS_BOOTSTRAP);
+            SECURITY_SECURITY_MODE);
         SOL_INT_CHECK_GOTO(r, < 0, err_exit);
 
+        switch (res[0].data[0].content.integer) {
+        case SOL_LWM2M_SECURITY_MODE_PRE_SHARED_KEY:
+            client->security = sol_lwm2m_client_security_add(client,
+                SOL_LWM2M_SECURITY_MODE_PRE_SHARED_KEY);
+            if (!client->security) {
+                r = -errno;
```

Comment:
`errno` will be positive (as it should, from what I understand) upon return from `sol_lwm2m_client_security_add()` [1], so in here we just need a negative `r` obtained from errno.

[1] https://github.com/bsmelo/soletta/blob/lwm2m-dtls_psk-v1/src/lib/comms/sol-lwm2m-security.c#L410